### ISSUE TITLE
Estimation Result Refactoring and adding String Resources

### ISSUE THREAD

**Issue by lifesbest23** [oripa/oripa#156]
I have started to implement a color chooser for the `EstimationResultFrame`, but @Ooouch1 was faster as in #155 it seems functionally done.
If you haven't started already I would add String resource usage to the `EstimationResultUI` Panel and refactor the class layout to resemble the `UIPanel` of the `MainFrame`?

I have been following some of your latest commits, and even if I do understand some of the implications of what you are working on, it would be interesting to know if these are changes you stumbled upon while trying to implement a certain change/issue or if you are generally refactoring the code base and this just happened to be a solvable issue?

**Comment by Ooouch1** [comment]
> If you haven't started already I would add String resource usage to the EstimationResultUI Panel and refactor the class layout to resemble the UIPanel of the MainFrame?

Yes you can update. I have merged moving `ColorRGBPanel` to `oripa.gui.view.util` package in #157 (thank you for the suggestion!) and I stop updating `EstimationResultUI` and around it.

> it would be interesting to know if these are changes you stumbled upon while trying to implement a certain change/issue or if you are generally refactoring the code base and this just happened to be a solvable issue?

First I work on interesting problem, then refactor the code base around. Recently I'm interested in performance issue which is figured out  in https://github.com/qurben/origami-editor. My ideas for speeding up algorithms has almost run out and I chose to implement color chooser because it was a solvable and practical issue. Sorry that I didn't leave some comment on issue #101 before starting implementation.

**Comment by lifesbest23** [comment]
Ok, I will probably finish the refactoring tomorrow.

Thanks for the link, I was not aware of another ORIHIME fork, at least I did not find it a year ago when I looked for Origami projects being actively developed. I have seen your commits regarding the folding optimizations, and you started to add some metrics to it. Do you know of a way to reliably measure the speed of certain functions used, and how to display these and the metrics? Is there something like unit testing for performance tests, or could they also be used for that?
Does origami-editor have similar performance issues or have they been able to solve them?

And I as well did not leave a comment on #101 and since you are the main contributor it should be my responsibility to make clear what I am working on :)
And switching between code optimization, refactoring, documenting and feature implementation is a good way to look at problems from new perspectives.

**Comment by Ooouch1** [comment]
> Do you know of a way to reliably measure the speed of certain functions used, and how to display these and the metrics?

I don't know much. VisualVM is a profiler for analyzing bottle neck of speed and memory but it is slow and can't be embedded in application.

> Does origami-editor have similar performance issues or have they been able to solve them?

They have solved. There are interesting articles by a committer of origami-editor:
https://origami.abstreamace.com/2021/10/13/fold-a-full-ryujin-3-5-with-orihime-algorithm/
https://origami.abstreamace.com/2021/11/01/fold-a-full-ryujin-3-5-with-orihime-algorithm-part-2/

Orihime couldn't finish folding kind of complex model in 1 hour according to the first article. After reading the article,  I tried my complex model on ORIPA and I got a similar situation. Now ORIPA's performance must be much better (but maybe not enough for Ryujin).

**Comment by lifesbest23** [comment]
Since I am working on a tiny display right now, I might have missed something, but for me everything still works.

Yeah I also found these two links, did not fully understand them though. When looking at the Origami-editor source it seems the have a really strict separation between view and model, for oripa it seems harder to distinguish, for me at least.

Is the file you have tried the one you shared in one of your recent issues or PR (I can't remember)?

Also might I suggest opening a separate issue to discuss performance optimization?

**Comment by Ooouch1** [comment]
I tried to keep models in domain.* packages, which include data entities and algorithms.

Yes, I use the CP I attached to #143 .

Feel free to open issues!

**Comment by lifesbest23** [comment]
#158 closes this.
